File: Codelearn/Traning/grasshopper.java
public class grasshopper {
    public static void main(String[] args) {
        System.out.println(grasshopper(80,2));
    }

    public static String grasshopper(long a,long b) {
        if(b == 1) return "1";
        long max = a/b;
        long count = 0;
        long permute = 1;
        for(long i = 0; i <= max; i++)
        {
            long totalJump = (a - i*b) + i;
            long sum = 1;

            for(long j = i - 1; j >= 0; j--)
            {
                sum *= (totalJump - j);
            }
            count += sum / permute;
            permute *= (i + 1);
            System.out.println(permute);
        }
        return count + "";
    }
}

File: Codelearn/Traning/grasshopper.js
function grasshopper(a, b){
    if(b === 1) return 1;
    let min = Math.floor(a/b);
    let f = [1];
    for(let i = 1; i < min; i++) f.push(f[i-1]*(i+1));
    
    let count = 1;
    for(let i = 1; i <= min; i++)
    {
        let numer = 1;
        let max = a - (i*b) + i;
        for(let j = i - 1; j >= 0; j--) numer *= (max-j);
        count += numer / f[i-1];
    }
    return count;
}